### ISSUE TITLE
weird full circle in arc

### ISSUE THREAD

**Issue by b0nb0n** [processing/p5.js#3949]
#### Nature of issue?

- [ x] Found a bug
- [ ] Existing feature enhancement
- [ ] New feature request

#### Most appropriate sub-area of p5.js?

- [ ] Color
- [ x] Core/Environment/Rendering
- [ ] Data
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Other (specify if possible)

#### Which platform were you using when you encountered this?

- [ ] Mobile/Tablet (touch devices)
- [ x] Desktop/Laptop
- [ ] Others (specify if possible)

#### Details about the bug: 

- p5.js version: 0.9.0
- Web browser and version: Google Chrome | 76.0.3809.100
- Operating System: Win10
- Steps to reproduce this:
I've unespected full circle in my sketch. This is not happened directly on processing.
You can try here: [https://codepen.io/atuco/pen/ExYyoWW](https://codepen.io/atuco/pen/ExYyoWW)

**Comment by limzykenneth** [comment]
When drawing the arc (I removed the `parseFloat()` as they are not necessary):

```javascript
arc(0, 0, sz, sz, pi, arcEnd);
```

if `pi` and `arcEnd` is at the same place, it will draw a circle (`pi` and `2*pi` is at the same place so using those two angles will draw a full circle). In your code, the value of `arcEnd` doesn't end up as `pi` (or the limited floating point representation of it since there isn't an infinite number of digits) but it gets really close to it.

There is a check in the code base if the two angles are really close together and if they are treat them as the same point and as such draw a full circle. The difference betweent the two angles needs to be greater than 0.00001 (as defined [here](https://github.com/processing/p5.js/blob/ca92fc457ff836e293916df06964e85f83182368/src/core/shape/2d_primitives.js#L43)) for them to be considered seperate points. 

We can reduce the value of `epsilon` (the name of the difference between the angles) but that makes it hard to draw a full circle by gradually increasing the value of `endArc` (ie. the opposite case of the sketch) which is why this was implemented like this in the first place.

@b0nb0n For your case, I would do something like this:

```javascript
arcEnd = map(sin(offSet+theta), -1, 1, pi+0.0001, 2*pi);
``` 
That way the value of `arcEnd` will not reach a point where the two angles are too close together while not having any change in how it looks.
